Update Remoting3 and exclude old HdrHistogram

datval opened the pull request:
Being on older versions causes dependency conflict inside atlasdb, which fails upgrades on atlas 0.66

Commit:
Update Remoting3 and exclude old HdrHistogram

schlosna commented:
Seems ok, do you have more details about the conflicts? I’m not sure the transitive excludes will get published and propagated in the way you’re expecting. We may need to publish a BOM for Tritium and AtlasDB to propagate desired dependency versions.

schlosna committed:
Improve Maven publishing via Gradle Nebula plugins

Publish BOM for tritium-lib and tritium-metrics modules to simplify
transitive dependency versioning.

schlosna committed:
Publish JavaDoc by default

schlosna committed:
Exclude duplicate dependencies from HdrHistogram metrics reservoir

schlosna committed:
Upgrade to Gradle 4.2.1

datval commented:
Thanks @schlosna,

This is the dependency tree of atlasdb-client which causes the problem
`
     |    |    |    |    |    +--- com.palantir.atlasdb:atlasdb-client:0.66.0
     |    |    |    |    |    |    +--- com.palantir.atlasdb:atlasdb-commons:0.66.0 (*)
     |    |    |    |    |    |    +--- com.palantir.atlasdb:atlasdb-api:0.66.0 (*)
     |    |    |    |    |    |    +--- com.palantir.atlasdb:atlasdb-client-protobufs:0.66.0
     |    |    |    |    |    |    |    \--- com.google.protobuf:protobuf-java:2.6.0
     |    |    |    |    |    |    +--- com.googlecode.json-simple:json-simple:1.1.1
     |    |    |    |    |    |    +--- commons-lang:commons-lang:2.6
     |    |    |    |    |    |    +--- org.xerial.snappy:snappy-java:1.1.1.7
     |    |    |    |    |    |    +--- com.googlecode.protobuf-java-format:protobuf-java-format:1.2
     |    |    |    |    |    |    +--- com.google.protobuf:protobuf-java:2.6.0
     |    |    |    |    |    |    +--- com.palantir.remoting3:tracing:3.5.1 (*)
     |    |    |    |    |    |    +--- com.palantir.safe-logging:safe-logging:0.1.3
     |    |    |    |    |    |    +--- com.palantir.tritium:tritium-lib:0.8.3
     |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-api:0.8.3
     |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-core:0.8.3
     |    |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-api:0.8.3
     |    |    |    |    |    |    |    |    \--- com.palantir.safe-logging:safe-logging:0.1.3
     |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-metrics:0.8.3
     |    |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-api:0.8.3
     |    |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-core:0.8.3 (*)
     |    |    |    |    |    |    |    |    +--- com.palantir.safe-logging:safe-logging:0.1.3
     |    |    |    |    |    |    |    |    +--- org.mpierce.metrics.reservoir:hdrhistogram-metrics-reservoir:1.1.2
     |    |    |    |    |    |    |    |    |    \--- org.hdrhistogram:HdrHistogram:2.1.9 -> 2.1.10
     |    |    |    |    |    |    |    |    \--- org.hdrhistogram:HdrHistogram:2.1.10
     |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-proxy:0.8.3
     |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-slf4j:0.8.3
     |    |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-api:0.8.3
     |    |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-core:0.8.3 (*)
     |    |    |    |    |    |    |    |    \--- com.palantir.safe-logging:safe-logging:0.1.3
     |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-tracing:0.8.3
     |    |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-api:0.8.3
     |    |    |    |    |    |    |    |    +--- com.palantir.tritium:tritium-core:0.8.3 (*)
     |    |    |    |    |    |    |    |    +--- com.palantir.remoting3:tracing:3.3.4 -> 3.5.1 (*)
     |    |    |    |    |    |    |    |    \--- com.palantir.safe-logging:safe-logging:0.1.3
     |    |    |    |    |    |    |    \--- com.palantir.safe-logging:safe-logging:0.1.3
     |    |    |    |    |    |    +--- org.hdrhistogram:HdrHistogram:2.1.10
     |    |    |    |    |    |    \--- com.squareup:javapoet:1.9.0
`

datval commented:
Can we merge this and release?
